clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -334,64 +344,192 @@ def _get_csv_files_for_alternative_exam_date(
 
 
 def create_scans_tsv_file(
-    input_path: Path,
+    bids_path: Path,
     clinical_data_dir: Path,
     clinical_specifications_folder: Path,
 ) -> None:
     """Create scans.tsv files for AIBL.
 
     Parameters
     ----------
-    input_path : Path
-        The path to the input folder.
+    bids_path : Path
+        The path to the BIDS folder.
 
     clinical_data_dir : Path
         The path to the directory to the clinical data files.
 
     clinical_specifications_folder : Path
         The path to the folder containing the clinical specification files.
     """
-    import glob
-    from os import path
-
-    import clinica.iotools.bids_utils as bids
 
-    specifications = _load_specifications(clinical_specifications_folder, "scans.tsv")
-    scans_fields = specifications[bids.StudyName.AIBL.value]
-    field_location = specifications[f"{bids.StudyName.AIBL.value} location"]
-    scans_fields_bids = specifications["BIDS CLINICA"]
-    fields_dataset = []
-    fields_bids = []
-
-    # Keep only fields for which there are AIBL fields
-    for i in range(0, len(scans_fields)):
-        if not pd.isnull(scans_fields[i]):
-            fields_bids.append(scans_fields_bids[i])
-            fields_dataset.append(scans_fields[i])
-
-    files_to_read = []
-    sessions_fields_to_read = []
-    for i in range(0, len(scans_fields)):
-        # If the i-th field is available
-        if not pd.isnull(scans_fields[i]):
-            file_to_read_path = clinical_data_dir / field_location[i]
-            files_to_read.append(glob.glob(str(file_to_read_path))[0])
-            sessions_fields_to_read.append(scans_fields[i])
-
-    bids_ids = [
-        path.basename(sub_path) for sub_path in glob.glob(str(input_path / "sub-AIBL*"))
-    ]
-    ses_dict = {
-        bids_id: {"M000": "bl", "M018": "m18", "M036": "m36", "M054": "m54"}
-        for bids_id in bids_ids
-    }
-    scans_dict = bids.create_scans_dict(
+    scans_dict = create_scans_dict(
         clinical_data_dir,
-        bids.StudyName.AIBL,
         clinical_specifications_folder,
-        bids_ids,
-        "RID",
-        "VISCODE",
-        ses_dict,
+        bids_path,
+    )
+    write_scans_tsv(bids_path, scans_dict)
+
+
+def _flutemeta_badline(line: List[str]) -> Optional[List[str]]:
```

Comment:
Sometimes I also forget about it, but since we are in the Python 3.9+ world, we can use builtins for type hints. I don't think we need to update all signatures in an industrial way, but slowly make the code evolve to use new typing syntax.

```suggestion
def _flutemeta_badline(line: list[str]) -> Optional[list[str]]:
```

Reply:
Okay, I will try to remember we can use it now